### PR TITLE
Modernise magicpen repository

### PR DESCRIPTION
**Pull request by alexjeffburke**
This PR brings the repository up-to-date and pulls in:
* eslint
* karma (ChromeHeadless)
* rollup

While here it also adds an ESM build of the module.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+trim_trailing_whitespace = true
+indent_size = 4
```
Comment (papandreou) [inline .editorconfig]: Would be nice to switch to 2 spaces and prettier while we're at it, but we can just follow up with that later.

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+module.exports = function(config) {
```
Comment (papandreou) [inline karma.conf.js]: Is it really necessary to test this module in Chrome Headless? Seems like some pretty heavy machinery to bring in.

Comment (alexjeffburke) [reply karma.conf.js]: Hm - maybe not. This module does need to run i. the browser, so on the basis it’s used that way by Unexpected I thought it made sense to cover that, but happy to go with this.

Comment (papandreou) [reply karma.conf.js]: My gut feeling says it's sufficient to cover that via the Unexpected test suite. Magicpen itself does not really integrate directly with browser apis.

The main reason I'm a bit hesitant to put browser tests into every module is that they tend to be fragile and add 100 MB worth of dev deps.

Comment (alexjeffburke) [reply karma.conf.js]: @papandrou that’s a good point - it doesn’t really use any of the native facilities. How would you feel if I made it run the browser build on node as more of a smoke test that it built properly?

Comment (papandreou) [reply karma.conf.js]: Sure :+1:

**Comment by sunesimonsen** [comment]
Released as magicpen@6.2.0

**Pull request merged**